cmd/analyzer/analyzer.go

```diff
@@ -329,14 +390,25 @@ func NewService(cfg *config.AnalysisConfig) (*Service, error) {
 	logger.Info("initialized all analyzers")
 
 	return &Service{
-		Analyzers: analyzers,
+		fastSyncAnalyzers: fastSyncAnalyzers,
+		analyzers:         analyzers,
 
 		sources: sources,
 		target:  dbClient,
 		logger:  logger,
 	}, nil
 }
 
+// closingChannel returns a channel that closes when the wait group `wg` is done.
+func closingChannel(wg *sync.WaitGroup) chan struct{} {
```

Comment:
```suggestion
func closingChannel(wg *sync.WaitGroup) <-chan struct{} {
```

That way a receive-only channel is returned which cannot be (mistakenly) closed.

Reply:
sweet, thank you